src/lib/methods/method.ts

```diff
@@ -16,4 +19,23 @@ export abstract class Method implements QueryBuilder {
 	 * Builds and returns the raw DynamoDB query object.
 	 */
 	abstract buildRawQuery(): any;
+
+	/**
+	 * Configure the retry policy
+	 *
+	 * @param opts - Retry configuration options.
+	 */
+	retry(retries: number | RetryOptions) {
+		this.retries = retries;
+
+		return this;
+	}
+
+	protected async runQuery<T>(operation: () => Promise<T>) {
+		const retries = this.retries || this.dynamodb.retries;
+
+		return retries
+			? retry(() => operation().catch(retryErrorHandler), {...typeof retries === 'number' ? {retries} : retries})
```

Comment:
Extracting the retry options to a variable will be easier to read I guess.

```ts
const retryOptions = typeof retries === 'number' ? {retries} : retries;
```

Reply:
I wasn't sure about it, but did we discuss passing default options to `p-retry`? Like the randomizer thing etc? I think we should come up with some sensible defaults.

Reply:
The current implementation uses the defaults from the `p-retry` library and the `retry` mechanism is an opt-in feature, it's not enabled by default.

Reply:
I know, but if we do `retry(5)`, the exponential back off thing should make sense and we should add some random overwriting the defaults from `retry`. Let's discuss tomorrow f2f :).